### PR TITLE
[mono] Remove duplicates from the AOT data blob.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by SamMonoRT** [comment]
what are the size savings here ? should we backport to release\6.0 ?

File: src/mono/mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -3095,6 +3096,47 @@ add_to_blob (MonoAotCompile *acfg, const guint8 *data, guint32 data_len)
...
+{
+	BlobItem *item1 = (BlobItem*)a;
+	BlobItem *item2 = (BlobItem*)b;
+
+	if (item1->len != item2->len || item1->align != item2->align)
+		return FALSE;
+	for (int i = 0; i < item1->len; ++i)
+		if (item1->data [i] != item2->data [i])
+			return FALSE;
+	return TRUE;
```
Comment (imhameed) [inline src/mono/mono/mini/aot-compiler.c:3128]: Why not `memcmp`?

**Comment by vargaz** [comment]
It saves about 250k on the System.Runtime tests on wasm. Probably don't need to backport.

**Review by imhameed — APPROVED** [review]

**Comment by vargaz** [comment]
Failures are unrelated.

**Pull request merged**